spec/truffle/interop/single_threaded_spec.rb

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+# 
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative '../../ruby/spec_helper'
+
+describe "Single-threaded mode" do
+
+  it "should give a helpful error if a thread is created" do
+    ruby_exe(
+      "Thread.new { }",
+      options: "-Xsingle_threaded=true",
+      args: '2>&1'
+    ).should =~ /threads not allowed in single-threaded mode/
+  end
+    
+  describe "allows native memory finalisers created" do
+    
+    it "before interop" do
+      ruby_exe(
+        "Truffle::FFI::Pointer.new(14); puts Polyglot.eval('ruby-single-threaded-test', '14')",
+        options: "-Xsingle_threaded=true"
+      ).should == "14\n"
+    end
+    
+    it "after interop" do
+      ruby_exe(
+        "Polyglot.eval('ruby-single-threaded-test', '14'); Truffle::FFI::Pointer.new(14); puts 14",
+        options: "-Xsingle_threaded=true"
+      ).should == "14\n"
+    end
+    
+  end
+  
+  describe "allows files to be read" do
+
+    it "before interop" do
+      ruby_exe(
+        "File.read('README.md'); Polyglot.eval('ruby-single-threaded-test', '14'); puts 14",
```

Comment:
How about using this file instead, i.e., `"File.read(#{__FILE__.inspect})`?

Reply:
Ok.